### PR TITLE
[multistage] [testing] Add transform operator unit test

### PR DESCRIPTION
**Pull request by 61yao**
1) Cover transform operator unit test
2) Add pre-check that transforms cannot be empty and the size has to match result schema size.

### CODE REVIEW

File: pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpression.java [hunk 1]
```diff
@@ -206,6 +206,7 @@ class FunctionCall implements RexExpression {
     private SqlKind _sqlKind;
     @ProtoProperties
     private FieldSpec.DataType _dataType;
+    // TODO: Make sure _functionName and _sqlKind are matching.
```
Comment (walterddr) [inline pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpression.java:209]: same as the other PRs. please remove this TODO

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/FunctionOperand.java [hunk 2]
```diff
@@ -29,7 +29,9 @@
 import org.apache.pinot.query.planner.logical.RexExpression;
 import org.apache.pinot.query.runtime.operator.OperatorUtils;
 
-
+/*
+   Functions that apply to TransformOperator.
+ */
```
Comment (walterddr) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/FunctionOperand.java:34]:
```suggestion
/*
 * FunctionOperands are generated from {@link RexExpression}s.
 */
```

File: pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/TransformOperatorTest.java [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    expectedRows = Arrays.asList(new Object[]{true, "str"}, new Object[]{true, "str"}, new Object[]{true, "str"});
+    Assert.assertEquals(resultRows.size(), expectedRows.size());
+    Assert.assertEquals(resultRows.get(0), expectedRows.get(0));
+    Assert.assertEquals(resultRows.get(1), expectedRows.get(1));
+    Assert.assertEquals(resultRows.get(2), expectedRows.get(2));
+  }
+
+  @Test(expectedExceptions = IllegalStateException.class, expectedExceptionsMessageRegExp = ".*transform operand "
+      + "should not be empty.*")
+  public void testWrongNumTransform() {
```
Comment (walterddr) [inline pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/TransformOperatorTest.java:225]: could you also add empty data block e.g. a block without any row?

**Review by walterddr — APPROVED** [review]
lgtm overall. please kindly take a look at the remaining comments

File: pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/TransformOperatorTest.java (hunk 3 above)
Comment (61yao) [reply pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/TransformOperatorTest.java:225]: I'll follow up in next PR.

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/FunctionOperand.java (hunk 2 above)
Comment (61yao) [reply pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/FunctionOperand.java]: done

**Pull request merged**